tmt/cli.py

```diff
@@ -1796,30 +1796,10 @@ def try_command(context: Context, image_and_how: str, arch: str, **kwargs: Any)
     """
 
     tmt.trying.Try.store_cli_invocation(context)
-
     # Inject custom image and provision method to the Provision options
-    if image_and_how:
-
-        # TODO: For now just pick the first image-how pair, let's allow
-        # specifying multiple images and provision methods as well
-        image_and_how = image_and_how[0]
-
-        # We expect the 'image' or 'image@how' syntax
-        matched = re.match("([^@]+)@([^@]+)", image_and_how.strip())
-        if matched:
-            options = {"image": matched.group(1), "how": matched.group(2)}
-        else:
-            options = {"image": image_and_how}
-
-        # Add guest architecture if provided
-        if arch:
-            options['arch'] = arch
-
-        # For 'connect' rename 'image' to 'guest'
-        if options.get('how') == 'connect':
-            options['guest'] = options.pop('image')
-
-        tmt.steps.provision.Provision.store_cli_invocation(context=None, options=options)
+    if context.params['image_and_how']:
```

Comment:
Why are the options stored only when `image_and_how` is defined? What is the point of this condition?

Reply:
Good question :) https://github.com/teemtee/tmt/commit/5b7f89205#diff-2bcd3056539a32d0370d652cfc6c04bd19b674cae84c341c3be7fc51cc3c60e6R1615

Reply:
Yeah, the thing is that you can do `tmt try --arch aarch64` and this would not work. And I think this is a perfectly valid use case. You are ok with the default provision method and image but just want to specify the arch.

Reply:
It would be good to check options first and then store them in the invocation if there's anything. Also dont' you want to actually use the `kwargs`?

```python
options = _construct_trying_provision_options(kwargs)
if options:
    ...
```

Reply:
Okay I edit it so we can specify just `arch` and I did use `context.params`, as for me it looks more readable

Reply:
Thanks, added a simple test case for this in 7e173580.